airflow/utils/context.py

```diff
@@ -178,16 +216,22 @@ class OutletEventAccessors(Mapping[str, OutletEventAccessor]):
     def __init__(self) -> None:
         self._dict: dict[str, OutletEventAccessor] = {}
 
+    def __str__(self) -> str:
+        return f"OutletEventAccessors(_dict={self._dict})"
+
     def __iter__(self) -> Iterator[str]:
         return iter(self._dict)
 
     def __len__(self) -> int:
         return len(self._dict)
 
-    def __getitem__(self, key: str | Dataset) -> OutletEventAccessor:
-        if (uri := coerce_to_uri(key)) not in self._dict:
-            self._dict[uri] = OutletEventAccessor({})
-        return self._dict[uri]
+    def __getitem__(self, key: str | Dataset | DatasetAlias) -> OutletEventAccessor:
+        dict_key = key.name if isinstance(key, DatasetAlias) else coerce_to_uri(key)
```

Comment:
Can we instead expand `coerce_to_uri` to also accept DatasetAlias instead? Also tying back to the common superclass comment above.

Reply:
True, but we don’t really care the output value is a URI here, just a key. Or other places this is used either. We can always rename the function.

Reply:
I tried to move the logic into `coerce_to_uri` and rename it as `extract_event_key`